### PR TITLE
[Azure Search 1] Add DatabaseOwnerFetcher to query for current owner information

### PR DESCRIPTION
**Pull request by joelverhagen**
Progress on https://github.com/NuGet/NuGetGallery/issues/6475.

Query inspired by:
https://github.com/NuGet/NuGet.Jobs/blob/master/src/Search.GenerateAuxiliaryData/SqlScripts/Owners.sql

Progress on https://github.com/NuGet/NuGetGallery/issues/6475

### CODE REVIEW

**Comment by joelverhagen** [comment]
There are two problems with unit tests here:

1. There's not much useful to assert with regards to the SQL string itself.
1. The factory returns `SqlConnection` which is not mockable.

Because of these hurdles and since the code is small, I chose not to unit test it.

File: src/NuGet.Services.AzureSearch/Owners2AzureSearch/IOwnerSetComparer.cs [hunk 1]
```diff
@@ -5,8 +5,18 @@
...
+    /// </summary>
     public interface IOwnerSetComparer
     {
+        /// <summary>
+        /// Compares two sets of owners to determine the package IDs that have changed. The returned dictionary
+        /// contains owners that were added, removed, and changed. For the "added" and "changed" cases, the owner set
+        /// is the new data. For the "removed" case, the set is empty.
+        /// </summary>
+        /// <param name="oldData">The old owner information, typically from storage.</param>
+        /// <param name="newData">The new owner information, typically from gallery DB.</param>
```
Comment (shishirx34) [inline src/NuGet.Services.AzureSearch/Owners2AzureSearch/IOwnerSetComparer.cs]: may be specify the return type of <id, list<owners>>?

**Comment by loic-sharma** [comment]
@joelverhagen The `azuresearch` branch was merged into the `dev` branch. I updated this to target the `dev` branch

File: src/NuGet.Services.AzureSearch/Owners2AzureSearch/IOwnerSetComparer.cs (hunk 1 above)
Comment (joelverhagen) [reply src/NuGet.Services.AzureSearch/Owners2AzureSearch/IOwnerSetComparer.cs]: Done.

**Pull request merged**